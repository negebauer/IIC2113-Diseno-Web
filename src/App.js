import React from "react"

class App extends React.Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <h2>Welcome to React 2</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    )
  }
}

export default App
