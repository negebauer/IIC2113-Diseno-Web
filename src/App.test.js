import React from "react"
import { shallow, mount } from "enzyme"
import App from "./App"

it("shallow renders without crashing", () => {
  shallow(<App />)
})

it("mount renders without crashing", () => {
  mount(<App />)
})

it("renders welcome message", () => {
  const wrapper = shallow(<App />)
  const welcome = <h2>Welcome to React</h2>
  expect(wrapper).toContainReact(welcome)
})
