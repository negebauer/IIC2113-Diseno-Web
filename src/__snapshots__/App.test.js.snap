// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Connect(App)
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  options={
    Object {
      "hydratation": Object {
        "blacklist": Array [
          "hydratation",
          "router",
        ],
      },
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    hydratation={
      Object {
        "done": true,
      }
    }
    hydrate={[Function]}
    options={
      Object {
        "hydratation": Object {
          "blacklist": Array [
            "hydratation",
            "router",
          ],
        },
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <ConnectedRouter
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <withRouter(Connect(Navigator))>
            <Route
              render={[Function]}
            >
              <Connect(Navigator)
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <Navigator
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  mail=""
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <div>
                    <nav>
                      <div
                        className="nav-wrapper"
                      >
                        <Link
                          className="brand-logo"
                          replace={false}
                          to="/"
                        >
                          <a
                            className="brand-logo"
                            href="/"
                            onClick={[Function]}
                          >
                            Logo
                          </a>
                        </Link>
                        <ul
                          className="right hide-on-med-and-down"
                          id="nav-mobile"
                        >
                          <li>
                            <Link
                              replace={false}
                              to="/signup"
                            >
                              <a
                                href="/signup"
                                onClick={[Function]}
                              >
                                Signup
                              </a>
                            </Link>
                          </li>
                          <li>
                            <Link
                              replace={false}
                              to="/login"
                            >
                              <a
                                href="/login"
                                onClick={[Function]}
                              >
                                Login
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                    </nav>
                    <Switch>
                      <Route
                        component={[Function]}
                        computedMatch={
                          Object {
                            "isExact": true,
                            "params": Object {},
                            "path": "/",
                            "url": "/",
                          }
                        }
                        exact={true}
                        location={
                          Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        path="/"
                      >
                        <styled.div>
                          <div
                            className="sc-gzVnrw ibZhkv"
                          >
                            <HelmetWrapper
                              defer={true}
                              encodeSpecialCharacters={true}
                            >
                              <SideEffect(NullComponent)
                                defer={true}
                                encodeSpecialCharacters={true}
                                title="IIC2113 Web"
                                titleAttributes={Object {}}
                              >
                                <NullComponent
                                  defer={true}
                                  encodeSpecialCharacters={true}
                                  title="IIC2113 Web"
                                  titleAttributes={Object {}}
                                />
                              </SideEffect(NullComponent)>
                            </HelmetWrapper>
                            <Route
                              component={[Function]}
                              computedMatch={
                                Object {
                                  "isExact": true,
                                  "params": Object {},
                                  "path": "/",
                                  "url": "/",
                                }
                              }
                              exact={true}
                              location={
                                Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                }
                              }
                              path="/"
                            >
                              <Connect(Home)
                                history={
                                  Object {
                                    "action": "POP",
                                    "block": [Function],
                                    "createHref": [Function],
                                    "go": [Function],
                                    "goBack": [Function],
                                    "goForward": [Function],
                                    "length": 1,
                                    "listen": [Function],
                                    "location": Object {
                                      "hash": "",
                                      "pathname": "/",
                                      "search": "",
                                      "state": undefined,
                                    },
                                    "push": [Function],
                                    "replace": [Function],
                                  }
                                }
                                location={
                                  Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  }
                                }
                                match={
                                  Object {
                                    "isExact": true,
                                    "params": Object {},
                                    "path": "/",
                                    "url": "/",
                                  }
                                }
                              >
                                <Home
                                  history={
                                    Object {
                                      "action": "POP",
                                      "block": [Function],
                                      "createHref": [Function],
                                      "go": [Function],
                                      "goBack": [Function],
                                      "goForward": [Function],
                                      "length": 1,
                                      "listen": [Function],
                                      "location": Object {
                                        "hash": "",
                                        "pathname": "/",
                                        "search": "",
                                        "state": undefined,
                                      },
                                      "push": [Function],
                                      "replace": [Function],
                                    }
                                  }
                                  location={
                                    Object {
                                      "hash": "",
                                      "pathname": "/",
                                      "search": "",
                                      "state": undefined,
                                    }
                                  }
                                  match={
                                    Object {
                                      "isExact": true,
                                      "params": Object {},
                                      "path": "/",
                                      "url": "/",
                                    }
                                  }
                                >
                                  <div
                                    className="container"
                                  >
                                    <h1>
                                      IIC2113 - Web
                                    </h1>
                                    <h3>
                                      Api is at url 
                                      NO_ENV
                                    </h3>
                                  </div>
                                </Home>
                              </Connect(Home)>
                            </Route>
                          </div>
                        </styled.div>
                      </Route>
                    </Switch>
                  </div>
                </Navigator>
              </Connect(Navigator)>
            </Route>
          </withRouter(Connect(Navigator))>
        </Router>
      </ConnectedRouter>
    </Provider>
  </App>
</Connect(App)>
`;
